packages/hardhat-core/src/internal/solidity/compilation-job.ts

```diff
@@ -130,7 +129,8 @@ function mergeCompilationJobs(
     }
   }
 
-  return flatten([...jobsMap.values()]);
+  // Array#flat This method defaults to depth limit 1
```

Comment:
This answers my previous question. Wow. I had no idea. Should we add a linter rule about not using `.flat` without an argument or something like that? This seems super error-prone.

Reply:
That'd be useful, but I can't find any rule for that

Reply:
Slightly OT: https://developer.chrome.com/blog/smooshgate/